pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -0,0 +1,607 @@
+		return errors.New("public key type is required")
+	}
+	if entry.PublicKey.PkixData == nil || len(entry.PublicKey.PkixData) == 0 {
+		return errors.New("public key pkix data is required")
+	}
+
+	p.entries[spireKeyID] = entry
+	return nil
+}
+
+func (p *Plugin) createKey(ctx context.Context, spireKeyID string, keyType keymanager.KeyType) (keyEntry, error) {
+	res := keyEntry{}
+	description := p.descriptionFromSpireKeyID(spireKeyID)
+	keySpec, ok := keySpecFromKeyType(keyType)
+	if !ok {
+		return res, newErrorf(codes.Internal, "unsupported key type: %v", keyType)
+	}
+
+	createKeyInput := &kms.CreateKeyInput{
+		Description:           aws.String(description),
+		KeyUsage:              types.KeyUsageTypeSignVerify,
+		CustomerMasterKeySpec: keySpec,
+	}
+
+	key, err := p.kmsClient.CreateKey(ctx, createKeyInput)
+	if err != nil {
+		return res, newErrorf(codes.Internal, "failed to create key: %v", err)
+	}
+
+	pub, err := p.kmsClient.GetPublicKey(ctx, &kms.GetPublicKeyInput{KeyId: key.KeyMetadata.KeyId})
+	if err != nil {
+		return res, newErrorf(codes.Internal, "failed to get public key: %v", err)
+	}
+
+	return keyEntry{
+		KMSKeyID: *pub.KeyId,
+		Alias:    p.aliasFromSpireKeyID(spireKeyID),
+		PublicKey: &keymanager.PublicKey{
+			Id:       spireKeyID,
+			Type:     keyType,
+			PkixData: pub.PublicKey,
+		},
+	}, nil
+}
+
+func (p *Plugin) fetchKeyEntry(ctx context.Context, aliasName string, spireKeyID string) (*keyEntry, error) {
+	describeResp, err := p.kmsClient.DescribeKey(ctx, &kms.DescribeKeyInput{KeyId: &aliasName})
+	if err != nil {
+		return nil, newErrorf(codes.Internal, "failed to describe key: %v", err)
+	}
+
+	if !describeResp.KeyMetadata.Enabled {
+		// this means something external to the plugin, deleted or disabled the key without removing the alias
+		// returning an error provides the opportunity or reverting this in KMS
+		return nil, newErrorf(codes.FailedPrecondition, "found disabled SPIRE key: %q, alias: %q", *describeResp.KeyMetadata.KeyId, aliasName)
+	}
+
+	if describeResp.KeyMetadata.KeyId == nil {
+		return nil, newErrorf(codes.FailedPrecondition, "found SPIRE alias without key: %q", aliasName)
+	}
+
+	keyType, ok := keyTypeFromKeySpec(describeResp.KeyMetadata.CustomerMasterKeySpec)
+	if !ok {
+		return nil, newErrorf(codes.Internal, "unsupported key spec: %v", describeResp.KeyMetadata.CustomerMasterKeySpec)
+	}
+
+	getPublicKeyResp, err := p.kmsClient.GetPublicKey(ctx, &kms.GetPublicKeyInput{KeyId: &aliasName})
+	if err != nil {
+		return nil, newErrorf(codes.Internal, "failed to get public key: %v", err)
+	}
+
+	return &keyEntry{
+		KMSKeyID: *describeResp.KeyMetadata.KeyId,
+		Alias:    aliasName,
+		PublicKey: &keymanager.PublicKey{
+			Id:       spireKeyID,
+			Type:     keyType,
+			PkixData: getPublicKeyResp.PublicKey,
+		},
+	}, err
+}
+
+func (p *Plugin) fetchAliasesPages(ctx context.Context) ([]*keyEntry, error) {
+	var keyEntries []*keyEntry
+	keyEntriesMutex := &sync.Mutex{}
```

Comment:
nit: this doesn't need to be a pointer
```suggestion
	var keyEntriesMutex sync.Mutex
```